api/bucketlist/bucketlist.py

```diff
@@ -0,0 +1,188 @@
+from api.auth.auth import auth
+from flask import request, abort, jsonify, g, Blueprint
+from api.models import Bucketlist, db, User, Item
+from flask_httpauth import HTTPTokenAuth
+from itsdangerous import TimedJSONWebSignatureSerializer as Serializer, SignatureExpired, BadSignature
+
+auth = HTTPTokenAuth(scheme='Bearer')
+
+bucket_view = Blueprint('bucket_view', __name__, url_prefix='/bucketlists')
+
+
+@auth.verify_token
+def verify_auth_token(token):
+    if not token:
+        return jsonify({"message": "supply token"}), 401
+    user_id = User.verify_auth_token(token=token)
+    if not user_id:
+        return False
+    g.user = db.session.query(User).filter_by(id=user_id).first()
+    return True
+
+
+@bucket_view.route("/", methods=["POST"])
+@auth.login_required
+def add_new_bucketlist():
```

Comment:
What happens if name is an empty string?
Consider having two steps for validation here or a function that checks if the values passed are not empty and also not there.e.g.:

```
def validate_data(value):
    if not value:
        return 'Value does not exist'
    elif value.length == 0:
       return 'Value must not be an empty string'

        


```